evalml/utils/gen_utils.py

```diff
@@ -277,19 +285,22 @@ def _convert_to_woodwork_structure(data):
     """
     Takes input data structure, and if it is not a Woodwork data structure already, will convert it to a Woodwork DataTable or DataColumn structure.
     """
+    ww_data = data
     if isinstance(data, ww.DataTable) or isinstance(data, ww.DataColumn):
-        return data
+        return ww_data
     # Convert numpy data structures to pandas data structures
     if isinstance(data, list):
-        data = np.array(data)
-    if isinstance(data, pd.api.extensions.ExtensionArray) or (isinstance(data, np.ndarray) and len(data.shape) == 1):
-        data = pd.Series(data)
-    elif isinstance(data, np.ndarray):
-        data = pd.DataFrame(data)
+        ww_data = np.array(data)
+
+    if isinstance(ww_data, pd.api.extensions.ExtensionArray) or (isinstance(ww_data, np.ndarray) and len(ww_data.shape) == 1):
+        ww_data = pd.Series(ww_data)
+    elif isinstance(ww_data, np.ndarray):
+        ww_data = pd.DataFrame(ww_data)
+
     # Convert pandas data structures to Woodwork data structures
-    if isinstance(data, pd.Series):
-        return ww.DataColumn(data)
-    return ww.DataTable(data)
+    if isinstance(ww_data, pd.Series):
+        return ww.DataColumn(ww_data)
+    return ww.DataTable(ww_data, copy_dataframe=True)
```

Comment:
Nice, I love the way this logic is organized @angela97lin ! Its easy to follow what transformations will be applied to the input data for each type of supported input.

I was reasoning through to see if we're subject to the same problem as what I highlighted in `_rename_column_names_to_numeric` above, where woodwork types aren't getting copied over. But I think we're all set here :)

Let's make sure we have solid test coverage of this method. I think we added that in a prev PR but can't remember -- @angela97lin would you mind linking to / describing that if so?